install/migrations/update_9.5.x_to_10.0.0/ticketcontract.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ * @var DB $DB
+ * @var Migration $migration
+ */
+
+$default_charset = DBConnection::getDefaultCharset();
+$default_collation = DBConnection::getDefaultCollation();
+
+if (!$DB->tableExists('glpi_ticketcontracts')) {
+   $query = "CREATE TABLE `glpi_ticketcontracts` (
+      `id` int NOT NULL AUTO_INCREMENT,
+      `tickets_id` int NOT NULL DEFAULT '0',
+      `contracts_id` int NOT NULL DEFAULT '0',
+      PRIMARY KEY (`id`),
+      UNIQUE KEY `unicity` (`tickets_id`,`contracts_id`),
+      -- KEY `tickets_id` (`tickets_id`),
```

Comment:
```suggestion
```

Reply:
I kept it because it wanted to discuss it with you.

Unit tests ask me to remove it because they field already exist in `unicity`.
However `contracts_id` is also in `unicity` and this time the unit test ask me to keep my separate key.

Why is it working this way ? IMO we either need both keys or none ?

Reply:
Nvm, forgot composite index work for the first key.